addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1433,14 +1434,22 @@ export class OdooEditor extends EventTarget {
         let start = range.startContainer;
         let end = range.endContainer;
         // Let the DOM split and delete the range.
+        const startBlock = closestBlock(start);
+        const endBlock = closestBlock(end);
+        const startLi = closestElement(start, 'li');
         const doJoin =
-            (closestBlock(start) !== closestBlock(range.commonAncestorContainer) ||
-            closestBlock(end) !== closestBlock(range.commonAncestorContainer))
-            && (closestBlock(start).tagName !== 'TD' && closestBlock(end).tagName !== 'TD');
+            (startBlock !== closestBlock(range.commonAncestorContainer) ||
+                endBlock !== closestBlock(range.commonAncestorContainer))
+            && (startBlock.tagName !== 'TD' && endBlock.tagName !== 'TD')
+            && (firstLeaf(startBlock) !== start || lastLeaf(endBlock) !== end);
```

Comment:
Add some explantory docString explaining this scenario why we are avoiding to doJoin to be true, just for the understanding sake.